release notes for CLI v1.8.0 (#1535)
astronomer/docs: astro/cli/release-notes.md

If you have any questions or a bug to report, contact [Astronomer support](https://cloud.astronomer.io/support).

## Astro CLI 1.8.0

> Release date: November 22, 2022

Reviewer: ```suggestion
Release date: November 23, 2022
```

## New `astro run` command

> You can now use the `astro run` command to compile and run a DAG without starting a complete Airflow environment. When you run the command, the CLI compiles your DAG and runs it in a single Airflow worker container based on your Astro project configurations. Running DAGs without a scheduler or webserver improves the speed at which you can develop and test DAGs.

Reviewer: ```suggestion
You can now use the `astro run` command to run and debug a DAG from the command line without starting a local Airflow environment. When you run the command, the CLI compiles your DAG and runs it in a single Airflow worker container based on your Astro project configurations. You can see task success or failure, as well as task logs, directly in your terminal.

This command is an alternative to running `astro dev restart` every time you make a change to your DAG. Running DAGs without a scheduler or webserver improves the speed at which you can develop and test data pipelines.
```

> For more information, see [Run and Debug DAGs with Astro Run](test-and-troubleshoot-locally.md#run-and-debug-dags-with-astro-run).

Reviewer: ```suggestion
To learn more, see [Run and Debug DAGs with Astro Run](test-and-troubleshoot-locally.md#run-and-debug-dags-with-astro-run).
```

## Additional improvements

> - When you run `astro deploy` with an empty `dags` folder, the CLI excludes your `dags` folder when building and pushing an image of your project to Astro. This lets you manage your DAGs and project images in separate repositories when using DAG-only Deployments.

Reviewer: ```suggestion
- When you run `astro deploy` with an empty `dags` folder, the CLI excludes your `dags` folder when building and pushing an image of your project to Astro. This lets you manage your DAGs and project images in separate repositories when using [DAG-only deploys](deploy-code.md#deploy-dags-only).
```

> - The `deployment inspect` command now includes a `dag-deploy-enabled` field, and the fields are now ordered by importance instead of alphabetical order.

Reviewer: ```suggestion
- The `deployment inspect` command now includes a `dag-deploy-enabled` field, and the fields are now ordered by logical groupings instead of alphabetical order.
```

### Bug fixes 

> - Fixed an issue where the changes specified in the `docker-compose.override.yaml` file stopped applying to users projects.

Reviewer: ```suggestion
- Fixed an issue where configurations specified in the `docker-compose.override.yaml` file of an Astro project were not properly applied. 
```

If you have any questions or a bug to report, contact [Astronomer support](https://cloud.astronomer.io/support).

## Astro CLI 1.8.0

Release date: November 22, 2022

> ## Run DAGs with `astro run`

Reviewer 2: ```suggestion
## New `astro run` command
```

> ## New Command `astro run` to Run DAGs

Reviewer 3: ```suggestion
## Run DAGs with `astro run`
```

> You can now use the `astro run` command to compile and run a DAG. When you run the command, your DAGs run within a Docker container built from your local `Dockerfile` that contains your DAGs, DAG utility files, python requirements, and environment variables from the `.env` file. This command simplifies the testing and debugging of DAGs locally.

Reviewer 2: ```suggestion
You can now use the `astro run` command to compile and run a DAG without starting a complete Airflow environment. When you run the command, the CLI creates a single Airflow worker container based on your Astro project configurations. Only running DAGs in a single container improves the speed at which you can develop and test DAGs.
```

> User can now compile and run a DAG with only the command line. `astro run` will run your DAGs within a Docker container built from your local `Dockerfile` containing your DAGs, DAG utility files, python requirements, and env vars from your `.env` file. This command will simplify the process of testing and debugging DAGs locally for many users.

Reviewer 3: ```suggestion
You can now use the `astro run` command to compile and run a DAG. When you run the command, your DAGs run within a Docker container built from your local `Dockerfile` that contains your DAGs, DAG utility files, python requirements, and environment variables from the `.env` file. This command simplifies the testing and debugging of DAGs locally.
```

> The command will automatically load Connections and Variables located in the [Airflow Settings file](develop-project.md#configure-airflow_settingsyaml-local-development-only). For more information read [Run and Debug DAGs with Astro Run](test-and-troubleshoot-locally.md#run-and-debug-dags-with-astro-run).

Reviewer 3: ```suggestion
The command automatically loads the connections and environment variables located in the [Airflow Settings file](develop-project.md#configure-airflow_settingsyaml-local-development-only). For more information, see [Run and Debug DAGs with Astro Run](test-and-troubleshoot-locally.md#run-and-debug-dags-with-astro-run).
```

> The command automatically loads the connections and environment variables located in the [Airflow Settings file](develop-project.md#configure-airflow_settingsyaml-local-development-only). For more information, see [Run and Debug DAGs with Astro Run](test-and-troubleshoot-locally.md#run-and-debug-dags-with-astro-run).

Reviewer 2: ```suggestion
For more information, see [Run and Debug DAGs with Astro Run](test-and-troubleshoot-locally.md#run-and-debug-dags-with-astro-run).
```

> ## New Command `organization audit-logs export`

Reviewer 2: ```suggestion
## New `astro organization audit-logs` command
```

> Organization admins can now export audit logs with the `astro organization audit-logs export —organization-name <name>` command. When you run the command, a GZIP file containing the audit logs is created.

Reviewer 2: ```suggestion
Organization admins can now export audit logs using `astro organization audit-logs`. When you run the command, the CLI creates a GZIP file containing audit logs.
```

> Organization admins can now export their orgs audit logs with `astro organization audit-logs export —organization-name <name>`. This command will create a gzip file of audit logs that users can unzip and parse.

Reviewer 3: ```suggestion
Organization admins can now export audit logs with the `astro organization audit-logs export —organization-name <name>` command. When you run the command, a GZIP file containing the audit logs is created.
```

## Additional improvements

> - If you run `astro deploy` with an empty `dag` folder the pushing of DAGs will be skipped and only your projects image will be updated. This enables you to manage your deployments DAGs and image in separate repos.

Reviewer 3: ```suggestion
- When you run `astro deploy` with an empty `dag` folder, the project image is updated, but the DAGs are not pushed. This behavior change lets you manage your deployment DAGs and project images in separate repositories.
```

> - When you run `astro deploy` with an empty `dags` folder, the project image is updated, but the DAGs are not pushed. This behavior change lets you manage your deployment DAGs and project images in separate repositories.

Reviewer 2: ```suggestion
- When you run `astro deploy` with an empty `dags` folder, the CLI excludes your `dags` folder when building and pushing an image of your project to Astro. This lets you manage your DAGs and project images in separate repositories when using DAG-only Deployments.
```
> This lets you manage your DAGs and project images in separate repositories using CI/CD.

Does it? Can you deploy a `dags` folder using the CLI if it's not contained in an Astro project?

> - The `deployment inspect` command now includes a `dag-deploy-enabled` field and the fields are now ordered by importance instead of alphabetical order.

Reviewer 2: ```suggestion
- The `deployment inspect` command now includes a `dag-deploy-enabled` field, and the fields are now ordered by importance instead of alphabetical order.
```

> - There are some minor improvements to the `deployment inspect` command. Specifically a `dag-deploy-enabled` field was added and the fields are now ordered by importance rather than alphabetical order.

Reviewer 3: ```suggestion
- The `deployment inspect` command now includes a `dag-deploy-enabled` flag and the flags are now ordered by importance instead of alphabetical order.
```
